src/sentry/static/sentry/app/views/workBatchDetailsWaitingToBeMerged/workbatchDetails.jsx

```diff
@@ -4,102 +4,153 @@ import withOrganization from 'app/utils/withOrganization';
     if (detailsId == null) {
+      return null;
+    }
+    let workDefinition = byIds[detailsId];
+    let {fields, buttons, id} = workDefinition;
+    if (fields === null || buttons === null || id === null) {
+      return null;
+    }
+    return workDefinition;
+  }
+
+  sendButtonClickedEvent(buttonEvent) {
+    this.props.sendButtonClickedEvent(this.props.organization, buttonEvent);
+  }
+
+  render() {
+    //TODO: merge this with files in the workBatchDetails folder
+    if (
+      this.props.workBatchDetailsEntry.loadingDetails ||
+      this.state === null ||
+      !this.state.fetched
+    ) {
       return <LoadingIndicator />;
     }
-    const workDefinition = byIds[detailsId];
-    let {buttons: buttonDefinitions, id: configId} = workDefinition;
-    for (const entry of buttonDefinitions) {
-      const buttonClick = () => {
-        this.sendButtonClickedEvent(entry.name, configId);
-      };
-      buttonsFromConfig.push(
-        <button
-          className="btn btn-sm btn-default"
-          onClick={buttonClick}
-          name={entry.name}
-        >
-          {entry.caption}
-        </button>
-      );
+    let workDefinition = this.fetchDefinitionsFromProps();
+    if (workDefinition === null) {
+      return <LoadingIndicator />;
     }
+    let {detailsId: workbatchId} = this.props.workBatchDetailsEntry;
     return (
-      <div>
-        <h1>Workbatch details proxy</h1>
-        {buttonsFromConfig}
-      </div>
+      <DetailsForm
+        workDefinition={workDefinition}
+        sendButtonClickedEvent={this.sendButtonClickedEvent}
+        handleChange={this.handleChange}
+        currentFieldValues={this.state.currentFieldValues}
+        workBatchId={workbatchId}
+      />
     );
   }
 
-  sendButtonClickedEvent(clickedButtonName, workBatchId) {
-    const buttonEvent = {
-      event: clickedButtonName,
-      workBatchId,
-    };
-    this.props.sendButtonClickedEvent(this.props.organization, buttonEvent);
+  initCurrentFieldValues() {
+    let {detailsId, byIds} = this.props.workBatchDetailsEntry;
+    let workbatch = byIds[detailsId];
+    let propertyNames = Object.keys(workbatch.properties);
+    let currentFieldValues = propertyNames.reduce((previous, attr) => {
+      return {
+        ...previous,
+        [attr]: workbatch.properties[attr].value,
+      };
+    }, {});
+    return new Promise((resolve) => {
+      this.setState((prevState) => {
+        return {
+          ...prevState,
+          currentFieldValues,
+        };
+      }, resolve);
+    });
+  }
+
+  handleChange(e) {
+    let {name, value} = e.target;
+    this.setState((prevState) => {
+      let {currentFieldValues} = {...prevState};
+      currentFieldValues[name] = value;
+      return {
+        currentFieldValues,
+      };
+    });
   }
 }
 
 WorkbatchDetails.propTypes = {
   ...ClimsTypes.List,
   organization: ClimsTypes.Organization.isRequired,
+  edvardtext: PropTypes.string,
```

Comment:
Should edvardtext be here? :)

Reply:
Hehe, I actually saw this after I sent in the PR, and wondered if it would be discovered. I'll remove it.